components/ExpenseForm/index.js

```diff
@@ -93,78 +23,108 @@ function ExpenseForm() {
-            <option value="0"> Select Category</option>
-            {data.map((category, index) => (
-              <option key={index} value={category._id}>
-                {category.name}
-              </option>
-            ))}
-          </StyledSelect>
-        </FormGroup>
-        <FormGroup>
-          <StyledTextarea
-            id="description"
-            name="description"
-            placeholder="Description"
-            rows="4"
-            maxLength={500}
-          />
-        </FormGroup>
-        <FormGroup>
-          <StyledInput
-            type="number"
-            id="amount"
-            name="amount"
-            placeholder="Amount"
-            min={0}
-            required
-          />
-        </FormGroup>
+      <FormTitle>{isEditMode ? "Edit Expense" : "Add Expense"}</FormTitle>
+      <form onSubmit={onSubmit}>
+        {isEditMode ? (
+          <>
+            <FormGroup>
+              <StyledInput
+                id="name"
+                name="name"
+                type="text"
+                placeholder="Title"
+                maxLength={100}
+                required
+                defaultValue={expense.name}
+              />
+            </FormGroup>
+            <FormGroup>
+              <StyledSelect
+                id="categoryId"
+                name="categoryId"
+                required
+                defaultValue={expense.categoryId[0]._id}
+              >
+                <option value="0"> Select Category</option>
+                {data.map((data, index) => (
+                  <option key={index} value={data._id}>
+                    {data.name}
+                  </option>
+                ))}
+              </StyledSelect>
+            </FormGroup>
+            <FormGroup>
+              <StyledTextarea
+                id="description"
+                name="description"
+                placeholder="Description"
+                rows="4"
+                maxLength={500}
+                defaultValue={expense.description}
+              />
+            </FormGroup>
+            <FormGroup>
+              <StyledInput
+                type="number"
+                id="amount"
+                name="amount"
+                placeholder="Amount"
+                min={0}
+                required
+                defaultValue={expense.amount}
+              />
+            </FormGroup>
+          </>
+        ) : (
+          <>
+            <FormGroup>
+              <StyledInput
+                id="name"
+                name="name"
+                type="text"
+                placeholder="Title"
+                maxLength={100}
+                required
+              />
+            </FormGroup>
+            <FormGroup>
+              <StyledSelect id="categoryId" name="categoryId" required>
+                <option value="0"> Select Category</option>
+                {data.map((data, index) => (
+                  <option key={index} value={data._id}>
```

Comment:
if you have access to the id as a unique identifier, better to use this for the key

```suggestion
                  <option key={data._id} value={data._id}>
```